tools/nnyeah/nnyeah/ConstructorTransforms.cs

```diff
@@ -173,23 +180,51 @@ void ReworkCodeBlockAsNeeded (IEnumerable<MethodDefinition> methods)
 
 		void ReworkCodeBlockAsNeeded (MethodDefinition method)
 		{
-			foreach ((int index, int argCount) in FindConstructorInstructionIndex (method.Body.Instructions)) {
+			var processor = method.Body.GetILProcessor ();
+			foreach ((Instruction instruction, int argCount) in FindConstructorInstruction (method.Body.Instructions)) {
 				switch (argCount) {
 					case 1:
-						method.Body.Instructions.Insert (index, Instruction.Create (OpCodes.Call, NativeHandleOpImplicit));
+						processor.InsertBefore (instruction, Instruction.Create (OpCodes.Call, NativeHandleOpImplicit));
 						break;
 					case 2: {
 						var variable = new VariableDefinition (BoolTypeDefinition);
 						method.Body.Variables.Add (variable);
-						method.Body.Instructions.Insert (index, Instruction.Create (OpCodes.Ldloc, variable));
-						method.Body.Instructions.Insert (index, Instruction.Create (OpCodes.Call, NativeHandleOpImplicit));
-						method.Body.Instructions.Insert (index, Instruction.Create (OpCodes.Stloc, variable));
+						processor.InsertBefore (instruction, Instruction.Create (OpCodes.Stloc, variable));
+						processor.InsertBefore (instruction, Instruction.Create (OpCodes.Call, NativeHandleOpImplicit));
+						processor.InsertBefore (instruction, Instruction.Create (OpCodes.Ldloc, variable));
 						break;
 					}
 					default:
 						throw new ConversionException (Errors.E0017, method, method.DeclaringType);
 				}
 			}
 		}
+
+		public bool TryGetConstructorCallTransformation (Instruction instruction, [NotNullWhen (returnValue: true)] out Transformation? result)
+		{
+			if (instruction.OpCode.Code == Mono.Cecil.Cil.Code.Newobj &&
+				instruction.Operand is MethodDefinition invokedMethod &&
+				TransformedConstructors.Contains (invokedMethod)) {
+				switch (invokedMethod.Parameters.Count) {
+				case 1:
+					result = new Transformation (null!, TransformationAction.Insert, new List<Instruction> () {
```

Comment:
If we are going to a low the transformation to take null, should we just make it nullable? Else the code in the transformation class is going to be written assuming it cannot be null when that is a lie.